dropwizard-index-page/src/main/java/com/palantir/indexpage/FileSystemIndexPage.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ */
+
+package com.palantir.indexpage;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.github.mustachejava.DefaultMustacheFactory;
+import com.github.mustachejava.Mustache;
+import com.google.common.base.Charsets;
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Files;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.StringWriter;
+import java.util.Map;
+
+/**
+ * Used to represent the index page loaded from the file system.
+ */
+public final class FileSystemIndexPage implements TemplatedAsset {
+
+    private final DefaultMustacheFactory factory;
+    private final File file;
+
+    private Map<String, String> context;
+    private long lastModified;
+    private Mustache mustacheCompiler;
+    private StringWriter writer;
+
+    public FileSystemIndexPage(File file) {
+        checkNotNull(file);
+
+        this.factory = new DefaultMustacheFactory();
+        this.file = file;
+        writer = new StringWriter();
+
+        try {
+            loadFile();
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("Index page cannot be found at " + file.toString(), e);
+        }
+    }
+
+    @Override
+    public Optional<String> getContent(Map<String, String> inputContext) {
+        checkNotNull(inputContext);
+
+        try {
+            checkAndReloadFile(inputContext);
+        } catch (FileNotFoundException e) {
+
+            // the file could be deleted during run time, so return absent in order to handle the behavior gracefully
+            return Optional.absent();
+        }
+
+        return Optional.of(writer.toString());
+    }
+
+    private synchronized void checkAndReloadFile(Map<String, String> inputContext) throws FileNotFoundException {
+
+        // If file has been changed, we need to load the file and resolve the template. Otherwise,
```

Comment:
it might be worth describing all 3 different scenarios in a javadoc on this method